src/utils/indexForSearch.ts

```diff
@@ -0,0 +1,42 @@
+import "dotenv/config"
+import { Connection, createConnection, getMongoRepository } from "typeorm"
+
+import { databaseConfig } from "../config/database"
+import { Collection } from "../Entities"
+import { search } from "../lib/search"
+
+/**
+ * This indexes an ElasticSearch cluster with collections data.
+ */
+
+export const indexForSearch = async () => {
+  let connection: Connection | undefined
+
+  try {
+    const connectionArgs = databaseConfig()
+    connection = await createConnection(connectionArgs)
+
+    if (connection.isConnected) {
+      const repository = getMongoRepository(Collection)
+
+      const collections = await repository.find()
+      for (const collection of collections) {
+        const keywords = collection.query.keyword
+        await search.client.index({
+          index: search.index,
+          type: "marketing_collection",
+          id: collection.id.toString(),
+          body: { name: collection.title, keywords },
```

Comment:
This schema will probably change. It seems as though Gravity's querying for ES will use `name`, `featured_names`, and `alternate_names` as the fields. So I imagine the keywords here should actually be placed into `alternate_names`.

Reply:
Aside from keywords. There are more properties you can add to the collection data payload that'll come in handy. Namely `query.artist_ids`, `query.tag_id`, `query.gene_ids`. This will allow us to do more complex querying such as the use case you asked about: looking up collections by `artist_id`.

Reply:
So, I'm not sure we need that...those are just using ElasticSearch as a giant denormalized data store (not the ideal use case, we do this a lot and it's an anti pattern).

Instead, I'm _just_ adding the text fields which will actually be using the text analysis + search functionality.

To answer questions like: what genes/artists/tags have associated collections, we should just allow querying KAWS with those fields, and add `collections` into those types w/ a `delegateSchema` call to KAWS.

Reply:
Right... that makes sense then. So maybe just consider `collection.category` (e.g. "Abstract Art")  which is also a text field ES can consider a search term & users could search collections by

Reply:
great call, will add that!